causalpy/skl_meta_learners.py

```diff
@@ -0,0 +1,489 @@
+from copy import deepcopy
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.linear_model import LogisticRegression
+from sklearn.utils import check_consistent_length
+from sklearn.model_selection import train_test_split
+
+from causalpy.utils import _is_variable_dummy_coded, _fit
+
+
+class MetaLearner:
+    "Base class for meta-learners."
+
+    def __init__(self, X: pd.DataFrame, y: pd.Series, treated: pd.Series) -> None:
+        # Check whether input is appropriate
+        check_consistent_length(X, y, treated)
+        if not _is_variable_dummy_coded(treated):
+            raise ValueError("Treatment variable is not dummy coded.")
+
+        self.cate = None
+        self.treated = treated
+        self.X = X
+        self.y = y
+        self.models = {}
+
+    def predict_cate(self, X: pd.DataFrame) -> np.array:
+        """
+        Predict out-of-sample conditional average treatment effect on given input X.
+        For in-sample treatement effect self.cate should be used.
+        """
+        raise NotImplementedError()
+
+    def predict_ate(self, X: pd.DataFrame) -> np.float64:
+        """
+        Predict out-of-sample average treatment effect on given input X. For in-sample treatement
+        effect self.ate() should be used.
+        """
+        return self.predict_cate(X).mean()
+
+    def ate(self):
+        "Returns in-sample average treatement effect."
+        return self.cate.mean()
+
+    def fit(self, X: pd.DataFrame, y: pd.Series, treated: pd.Series, coords=None):
+        "Fits model."
+        raise NotImplementedError()
+
+    def summary(self):
```

Comment:
Shall we output a dataframe ? I imagine we would like to do some plots with the output.